### ISSUE TITLE
Invalid argument: model input must specify 'dims'

### ISSUE THREAD

**Issue by ligonzheng** [NVIDIA/tensorrt-inference-server#294]
I have a tensorflow savedmodel and deploy it with trtis.  I didn't set any config file but I add the --strict-model-config=false option. After deployment ,The status is as blow:
id: "inference:0"
version: "1.0.0"
uptime_ns: 128825324092
model_status {
  key: "inception_new"
  value {
    config {
      name: "inception_new"
      platform: "tensorflow_savedmodel"
      version_policy {
        latest {
          num_versions: 1
        }
      }
      max_batch_size: 1
      input {
        name: "input"
        data_type: TYPE_FP32
        dims: 299
        dims: 299
        dims: 3
      }
      input {
        name: "keep_prob"
        data_type: TYPE_FP32
      }
      input {
        name: "is_training"
        data_type: TYPE_BOOL
      }
      output {
        name: "output"
        data_type: TYPE_FP32
        dims: 5
      }
      instance_group {
        name: "inception_new"
        count: 1
        gpus: 0
        kind: KIND_GPU
      }
      default_model_filename: "tf_savedmodel"
    }
    version_status {
      key: 1
      value {
        ready_state: MODEL_UNAVAILABLE
      }
    }
  }
}
status_stats {
  success {
    count: 4088
    total_time_ns: 41681841
  }
}

and the error is : Invalid argument: model input must specify 'dims'
Model is unavailable, How to overcome this problem ?
I know if I redefine my tensorflow graph and remove those tensor like is_training would be a solution, But it's a big cost. Is there any other good way ? If I add a config file and add reshape config will work ?

**Comment by CoderHam** [comment]
Specify dims as an **array** if more than one dimension ->   e.g.:  ```dims: [ 3, 224, 224 ]```

**Comment by CoderHam** [comment]
@ligonzheng, our current model config auto-fill does not support control flags of the form **keep_prob** and  **is_training**. Your current work around would be to create a custom config file with the **keep_prob** and  **is_training** defined with ```dims: 1```

This also means that you cannot use dynamic batching (due to the same assumptions made while parsing the config model. Once you turn off _dynamic batching_ and define the _dims: 1_ it should work.
  
Let me know if otherwise!

**Comment by ligonzheng** [comment]
I had try your method but the result is not good. I added a config.pbtxt like this:
name: "inception_new"
platform: "tensorflow_savedmodel"
version_policy {
latest {
num_versions: 1
}
}
max_batch_size: 128
input [
{
name: "input"
data_type: TYPE_FP32
dims: [299,299,3]
},
{
name: "keep_prob"
data_type: TYPE_FP32
dims: [1]
},
{
name: "is_training"
data_type: TYPE_BOOL
dims: [1]
}
]
output [
{
name: "output"
data_type: TYPE_FP32
dims: [5]
}
]
As you said ,I added additional dims for is_training and keep_prob , no dynamic batching was setted. but I got this error if I use 19.04-py3: 
 Internal: unable to load model 'inception_new', model configuration supports batching but first dimension of tensor 'keep_prob' expected by framework is not a variable-size batch dimension: []
I got another kind of error if I use 19.03-py3:
Invalid argument: unable to load model 'inception_new', input 'keep_prob' dims [] don't match configuration dims [1]

**Comment by CoderHam** [comment]
@ligonzheng could you try this with the latest release -> **19.05**-py3 ? I believe the support for such model configurations were incorporated in that release.

**Comment by deadeyegoodwin** [comment]
@ligonzheng Your model should be supported by TRTIS if you use the master branch (or 19.05 when it is released later this week). The problem seems to be that your model requires "reshape" which is available in 19.04 but the autofill (--strict-model-config=false) support for reshape was not added until 19.05.

Can you try loading your model (using --strict-model-config=false) with 19.05 (or master) and let us know if your problem is fixed.

**Comment by ligonzheng** [comment]
@deadeyegoodwin Do you mean I can  use the autofill and a config file at the same time in 19.05?  otherwise how should I add the reshape configuration with the --strict-model-config=false option

**Comment by deadeyegoodwin** [comment]
Closing. @ligonzheng 19.05 just released. If you still see a problem with 19.05 please reopen this issue.

**Comment by ligonzheng** [comment]
I tried my model again with 19.04 and 19.05. Different settings were made when I generate the tf savedmodel.  The cases are as below:
case 1, Reshape is_training tensor and keep_prob tensor to [-1,1], the /api/status shows the model is available. I try to access the server, it returns "You must feed a value for placeholder tensor 'keep_prob with dtype float'".
case 2, No reshape operation for is_training and keep_prob. The trtis stopped and return "model input cannot have empty reshape for non-batching model". This case is like the title which using 19.03.
case 3, Reshape is_training tensor and keep_prob tensor to [1,], th /api/status shows the model is available but the input tensor for picture data has a dim of -1. I tried to access the server, it returns "model supports variable size for input 'input', request must specify input shape for model".
case 4,remove the keep_prob and is_training when I define the signature. The /api/status shows the model is available. I tried to access the server, it returns "You must feed a value for placeholder tensor 'keep_prob with dtype float'".
`def convert(model_path):
    with tf.Session() as sess:
        saver = tf.train.import_meta_graph(model_path + '.meta')
        saver.restore(sess, model_path)
        graph = tf.get_default_graph()

        #writer = tf.summary.FileWriter("./log_inception",sess.graph)
        #writer.close()
        input_tensor = graph.get_tensor_by_name('input:0')
        output_tensor = graph.get_tensor_by_name('InceptionV3/Predictions/Softmax:0')
        is_training = graph.get_tensor_by_name("is_training:0")
        keep_prob = graph.get_tensor_by_name("keep_prob:0")

        is_training = tf.reshape(is_training,[None,1])
        keep_prob = tf.reshape(keep_prob,[None,1])
        #tf.graph_util.remove_training_nodes(input_graph=graph)

        new_path = "inception_new/1"
        builder = tf.saved_model.builder.SavedModelBuilder(new_path)

        inputs = {'input': tf.saved_model.utils.build_tensor_info(input_tensor),'is_training': tf.saved_model.utils.build_tensor_info(is_training),'keep_prob':tf.saved_model.utils.build_tensor_info(keep_prob)}
        outputs = {'output':tf.saved_model.utils.build_tensor_info(output_tensor)}
        method_name = tf.saved_model.signature_constants.PREDICT_METHOD_NAME

        my_signature = tf.saved_model.signature_def_utils.build_signature_def(inputs,outputs,method_name)
        builder.add_meta_graph_and_variables(sess,['serve'],signature_def_map={'serving_default':my_signature})
        builder.save()
`
